compiler/rustc_mir_build/src/build/scope.rs

```diff
@@ -618,6 +618,16 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             }
         } else {
             assert!(value.is_none(), "`return` and `break` should have a destination");
+            // `continue` statements generate no MIR statement with the `continue` statement's Span,
+            // and the `InstrumentCoverage` statement will have no way to generate a coverage
+            // code region for the `continue` statement, unless we add a dummy `Assign` here:
+            let mut local_decl = LocalDecl::new(self.tcx.mk_unit(), span);
+            local_decl = local_decl.immutable();
+            let temp = self.local_decls.push(local_decl);
+            let temp_place = Place::from(temp);
+            self.cfg.push(block, Statement { source_info, kind: StatementKind::StorageLive(temp) });
+            self.cfg.push_assign_unit(block, source_info, temp_place, self.tcx);
+            self.cfg.push(block, Statement { source_info, kind: StatementKind::StorageDead(temp) });
```

Comment:
If those storage statements are only to pass the validation in the state transform, the local could be marked as internal instead.

Reply:
Oh, I'll try that.

Reply:
Great! It looks like that worked. I did have to drop the `immutable()`, otherwise, at least with multiple `continue` statements in the function, I got:
```
error[E0384]: cannot assign twice to immutable variable value
  --> src/test/run-make-fulldeps/coverage/continue.rs:10:17
   |
10 |                 continue;
   |                 ^^^^^^^^ cannot assign twice to immutable variable
```

This works without calling `immutable()` though. Cleaner!